migration-engine/connectors/sql-migration-connector/src/sql_database_migration_inferrer.rs

```diff
@@ -81,11 +97,57 @@ fn infer(
     current_database_schema: SqlSchema,
     expected_database_schema: SqlSchema,
     flavour: &dyn SqlFlavour,
+    next_datamodel: &Datamodel,
 ) -> SqlMigration {
     let steps =
         sql_schema_differ::calculate_steps(Pair::new(&current_database_schema, &expected_database_schema), flavour);
+    let next_schema = &expected_database_schema;
+
+    let added_columns_with_virtual_defaults: Vec<(usize, usize)> = steps
+        .iter()
+        .filter_map(|step| step.as_alter_table())
+        .flat_map(move |alter_table| {
+            alter_table
+                .changes
+                .iter()
+                .filter_map(|change| change.as_add_column())
+                .map(move |column| -> (usize, usize) { (*alter_table.table_index.next(), column.column_index) })
+        })
+        .chain(
+            steps
+                .iter()
+                .filter_map(|step| step.as_redefine_tables())
+                .flat_map(|redefine_tables| redefine_tables.into_iter())
+                .flat_map(move |table| {
+                    table
+                        .added_columns
+                        .iter()
+                        .map(move |column_index| (*table.table_index.next(), *column_index))
+                }),
+        )
+        .map(|(table_index, column_index)| {
+            let table = next_schema.table_walker_at(table_index);
+            let column = table.column_at(column_index);
+
+            (table, column)
+        })
+        .filter(|(table, column)| {
+            walk_models(next_datamodel)
+                .find(|model| model.database_name() == table.name())
+                .and_then(|model| model.find_scalar_field(column.name()))
+                .filter(|field| {
+                    field
+                        .default_value()
+                        .map(|default| default.is_uuid() || default.is_cuid())
+                        .unwrap_or(false)
+                })
+                .is_some()
+        })
+        .map(move |(table, column)| (table.table_index(), column.column_index()))
+        .collect();
```

Comment:
Sometimes I wonder if we wouldn't need a query language for our schema datastructures.

Reply:
maybe a lisp?